### DISCUSSION TITLE
WebView, PropertyGrid, Ribbon controls

### DISCUSSION THREAD

**Discussion by kinsung** [AvaloniaUI/Avalonia#16147] [Ideas]
Consider developing centralized controls that are essential for modern software development, including: WebView, PropertyGrid, Ribbon controls.

The following url is for reference:
https://github.com/VitalElement/CefGlue.Core/tree/master/CefGlue.Avalonia
https://github.com/amazerol/AvaloniaRibbon

**Comment by Splitwirez** [comment]
@kinsung FYI development of the Ribbon you linked has pretty much moved to https://github.com/Splitwirez/AvaloniaRibbon at this point

**Comment by Ruluk** [comment]
Possible suggestion: Integrate with https://github.com/webview/webview to use the native webview instead of bundling CEF?

**Comment by maxkatz6** [comment]
@Ruluk if it's possible to get native control handle from this library, than it could be easily wrapped with NativeControlHost.
See native controls example:  https://github.com/AvaloniaUI/Avalonia/blob/master/samples/interop/NativeEmbedSample/EmbedSample.cs#L77-L91

**Comment by IgorIIV** [comment]
Hey, guys.
Are there any plans to implement PropertyGrid? Very much needed control.

**Comment by maxkatz6** [comment]
PropertyGrid is unlikely to be part of Avalonia main package.
But you can find third party controls, like https://github.com/mameolan/Avalonia.ExtendedToolkit/tree/master/Avalonia.ExtendedToolkit/Controls/PropertyGrid

**Comment by IgorIIV** [comment]
Not bad. Thank you.

**Comment by bodong1987** [comment]
i create a simple one here: https://github.com/bodong1987/Avalonia.PropertyGrid

**Comment by timunie** [comment]
@bodong1987 looks beautiful 👍. Probably you want to add it to https://github.com/AvaloniaCommunity/awesome-avalonia#controls ?

**Comment by timdinh** [comment]
I think webview support is essential because if there is any controls or UI element that we need but not natively supported yet we can just use webview to render our UI as a workaround

**Comment by maxkatz6** [comment]
From this list, I don't think we will ever have PropertyGrid and Ribbon build-in.
Native WebView is something that could happen at some point, but it's very problematic cross-platform (specifically on Linux), making it, in general, easier to use CEF-based solutions, which there are already multiple for Avalonia.

**Comment by Splitwirez** [comment]
> I think webview support is essential because if there is any controls or UI element that we need but not natively supported yet we can just use webview to render our UI as a workaround

@timdinh If they don't exist, you'd have to write them either way, so why not write them for Avalonia directly? (I'm missing something here, aren't I?)

-----

> From this list, I don't think we will ever have [...] Ribbon build-in.

@maxkatz6 Makes me wish I wasn't so flaky about maintaining my own Ribbon (third-party lib)...

> Native WebView is something that could happen at some point, but it's very problematic cross-platform (specifically on Linux), making it, in general, easier to use CEF-based solutions, which there are already multiple for Avalonia.

[Why is everything...Chrome?](https://reddit.com/r/ProgrammerHumor/comments/kzts4s/welcome_to_the_future_old_man/)
